### ISSUE TITLE
Can't use `OidcSecurityService` from an APP_INITIALIZER

### ISSUE THREAD

**Issue by ch4mpy** [damienbod/angular-auth-oidc-client#1268]
**Describe the bug**
Apparently, configuration injected when importing AuthModule is not retrieved correctly when using `OidcSecurityService` from an APP_INITILIZER.

When I try to use `oidcSecurityService.checkAuth()` in an `APP_INITIALIZER`, I get 
```
core.js:6479 ERROR TypeError: Cannot read property 'configId' of null
    at OidcSecurityService.getConfiguration (angular-auth-oidc-client.js:4466)
```

**To Reproduce**
1. Create an UaaService like this one
``` typescript
@Injectable({ providedIn: 'root' })
export class UaaService {
    constructor(private oidcClient: OidcSecurityService) {}

    async init(): Promise<any> {
        await this.oidcClient.checkAuth().toPromise()
        // more initialisation
    }

    // more OidcSecurityService wrapping
}
```
2. Add an APP_INITILIZER to auth module to call UaaService init method:
``` typescript
export function initUaaService(uaaService: UaaService) {
    return () => uaaService.init()
}

@NgModule({
    imports: [
        AuthModule.forRoot({
            config: environment.openIdConfig,
        }),
    ],
    declarations: [],
    exports: [],
    providers: [
        {
            provide: APP_INITIALIZER,
            useFactory: initUaaService,
            deps: [UaaService],
            multi: true,
        },
    ],
})
export class RimaAuthModule {}
```
3. load application

**Expected behavior**
OidcSecurityService.checkAuth() is awaited during Angular app initialization so that a wrapping service is initialised with retrieved identity

**Comment by FabianGosebrink** [comment]
What are you trying to do? Maybe the guards are a better approach?

**Comment by ch4mpy** [comment]
@FabianGosebrink I'm trying to initialise a service with a dependency on `OidcSecurityService`, during app initialisation (before first guard is evaluated).

This service is mostly a wrapper around `OidcSecurityService` that exposes, among other stuff, a typed users observable. Currently, this observable is a BehaviorSubject initialised with an anonymous user and updated after `OidcSecurityService` emits a `userData$`. 

My problem is thatmost of the time (not even always),  if I call `this.oidcSecurityService.checkAuth().toPromise()` from `AppComponent.onInit()`, OidcSecurityService initialisation is not finished when first guard is evaluated.

So basically, I really want to await checkAuth in an APP_INITIALIZER to have a fully initialised security context when the first route is evaluated and component initialised.

**Comment by ph3b** [comment]
Having the same issue after ugprading from v11. I'm doing this because I want to sync userData$ with my ngrx store. Have you found any alternative ways of doing this?

**Comment by FabianGosebrink** [comment]
> My problem is that most of the time (not even always), if I call this.oidcSecurityService.checkAuth().toPromise() from AppComponent.onInit(), OidcSecurityService initialisation is not finished when first guard is evaluated.

That is what the `AutoLoginAllRoutesGuard` is trying to solve. So I think updating may be worth a try.

**Comment by MarinichViktor** [comment]
Have similar issue. I solved it by using ConfigLoaded and ConfigLoadingFailed events from PublicEventsService 
`  {
    provide: APP_INITIALIZER,
    useFactory: (events: PublicEventsService) => {
      return () => events.registerForEvents().pipe(
          filter(({type}) => [EventTypes.ConfigLoaded, EventTypes.ConfigLoadingFailed].includes(type)
          first(),
        ),
        mergeMap(() => myInitFunction()),
      );
    },
    deps: [
      PublicEventsService,
    ],
    multi: true,
  }``

**Comment by vixisoft** [comment]
i've been scratching my head around because @MarinichViktor solution didn't work for me!
The events are not firing.

I've come to a solution to this problem that works well for now,

I'm using =>    "angular-auth-oidc-client": "12.0.3",

```
{ provide: APP_INITIALIZER, useFactory: init, deps: [OidcSecurityService], multi: true  },

function init(oidcSecurityService: OidcSecurityService): Function {
    return () => new Promise<LoginResponse>((resolve) => {
        setTimeout(() => {
            oidcSecurityService.checkAuth().subscribe(resolve);
        });
    });
}
```
